src/backend/distributed/catalog/dependency.c

```diff
@@ -215,15 +278,35 @@ OrderObjectAddressListInDependencyOrder(List *objectAddressList)
  */
 static void
 recurse_pg_depend(const ObjectAddress *target,
-				  bool (*follow)(void *context, const Form_pg_depend row),
-				  void (*apply)(void *context, const Form_pg_depend row),
+				  List * (*expand)(void *context, const ObjectAddress *target),
+				  bool (*follow)(void *context, Form_pg_depend row),
+				  void (*apply)(void *context, Form_pg_depend row),
 				  void *context)
 {
 	Relation depRel = NULL;
 	ScanKeyData key[2] = { 0 };
 	SysScanDesc depScan = NULL;
 	HeapTuple depTup = NULL;
 
+	/********************
+	* first apply on expanded entries of pg_depend
+	********************/
+	if (expand != NULL)
```

Comment:
As discussed, let's capture the contents of the table and concat the expanded entries in a list and iterate over it.